query_executor.go

```diff
@@ -75,34 +62,28 @@ func (q *queryExecutor) executeQuery(qry ExecutableQuery) (*Iter, error) {
 			select {
 			case <-ticker.C:
 				// Launch the additional execution
-				specWG.Add(1)
-				go q.run(qry, &specWG, results, stop)
-			case <-qry.GetContext().Done():
-				// not starting additional executions
-				return
-			case <-stop:
+				go q.run(ctx, qry, results)
+			case <-ctx.Done():
 				// not starting additional executions
 				return
 			}
 		}
 	}()
 
-	res := <-results
-	if res.iter == nil && res.err == nil {
-		// if we're here, the results channel was closed, so no more hosts
-		return nil, ErrNoConnections
+	select {
+	case iter := <-results:
+		return iter, nil
```

Comment:
There isnt any point as we have either sent a result or the context is already canceled. The context here is canceled when executeQuery returns so that means it must have read from the result channel.

We don't need to close the channel no, any sends to a closed channel panic and all sends already return due to the context being cancelled on return.